cocos/core/math/color.ts

```diff
@@ -734,43 +735,10 @@ export class Color extends ValueType {
      * @param other The specified color.
      */
     public multiply (other: Color): Color {
-        const r = ((this._val & 0x000000ff) * other.r) >> 8;
-        const g = ((this._val & 0x0000ff00) * other.g) >> 8;
-        const b = ((this._val & 0x00ff0000) * other.b) >> 8;
-        const a = ((this._val & 0xff000000) >>> 8) * other.a;
-        this._val = (a & 0xff000000) | (b & 0x00ff0000) | (g & 0x0000ff00) | (r & 0x000000ff);
-        return this;
-    }
-
-    /**
-     * @deprecated since v3.5.0, this is an engine private interface that will be removed in the future.
-     */
-    public _set_r_unsafe (red: number): Color {
-        this._val = ((this._val & 0xffffff00) | red) >>> 0;
-        return this;
-    }
-
-    /**
-     * @deprecated since v3.5.0, this is an engine private interface that will be removed in the future.
-     */
-    public _set_g_unsafe (green: number): Color {
-        this._val = ((this._val & 0xffff00ff) | (green << 8)) >>> 0;
-        return this;
-    }
-
-    /**
-     * @deprecated since v3.5.0, this is an engine private interface that will be removed in the future.
-     */
-    public _set_b_unsafe (blue: number): Color {
-        this._val = ((this._val & 0xff00ffff) | (blue << 16)) >>> 0;
-        return this;
-    }
-
-    /**
-     * @deprecated since v3.5.0, this is an engine private interface that will be removed in the future.
-     */
-    public _set_a_unsafe (alpha: number): Color {
-        this._val = ((this._val & 0x00ffffff) | (alpha << 24)) >>> 0;
+        this.r *= other._r;
+        this.g *= other._g;
+        this.b *= other._b;
+        this.a = other._a;
```

Comment:
Why doesn't `a` need to be `multiplied`.

Reply:
fixed